### PR TITLE
Fix RuboCop for Casks.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

---

Split out RuboCop config used for Casks.

### CODE REVIEW

File: Library/.rubocop_shared.yml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+AllCops:
```
Comment (MikeMcQuaid) [inline Library/.rubocop_shared.yml]: What’s the point of this file? Given the cask file already duplicates much it seems strange to inherit to save the few lines this does when running ‘rubocop’ outside of ‘brew cask style’ in an editor will return wrong results again.

File: Library/.rubocop_cask.yml [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+  Description: 'Ensure that cask stanzas are grouped correctly. More info at https://github.com/Homebrew/homebrew-cask/blob/master/CONTRIBUTING.md#stanza-order'
+  Enabled: true
+
+Cask/StanzaOrder:
+  Description: 'Ensure that cask stanzas are sorted correctly. More info at https://github.com/Homebrew/homebrew-cask/blob/master/CONTRIBUTING.md#stanza-order'
+  Enabled: true
+
+# Built-In Cops
+
+Layout/AlignHash:
```
Comment (MikeMcQuaid) [inline Library/.rubocop_cask.yml]: A bunch of these values are identical to rubocop.yml or rubocop defaults. I spent a bunch of time checking both.

File: Library/.rubocop_shared.yml (hunk 1 above)
Comment (reitermarkus) [reply Library/.rubocop_shared.yml]: > when running ‘rubocop’ outside of ‘brew cask style’ in an editor will return wrong results again.

It doesn't return wrong results, each tap has a config file inheriting from `.rubocop_cask.yml`.

File: Library/.rubocop_cask.yml (hunk 2 above)
Comment (reitermarkus) [reply Library/.rubocop_cask.yml]: I will take another look and move the duplicated rules into `.rubocop_shared.yml`.

**Pull request merged**